arraycontext/fake_numpy.py

```diff
@@ -250,7 +274,7 @@ def norm(self, ary, ord=None):
                 return flat_norm(ary, ord=ord)
 
         if is_array_container(ary):
-            return _scalar_list_norm([
```

Comment:
Deleting the unused `_scalar_list_norm` is probably going to fail all the downstream tests, because `meshmode` imports it :( (which is definitely my bad). 

That's why it's copy pasted into inducer/meshmode#273, so we can remove it here.